core/src/stored/backends/win32_tape_device.cc

```diff
@@ -265,20 +265,20 @@ int win32_tape_device::d_open(const char* pathname, int, int)
   return (int)idxFile + 3;
 }
 
-ssize_t win32_tape_device::d_read(int fd, void* buffer, size_t count)
+ssize_t win32_tape_device::d_read(int t_fd, void* buffer, size_t count)
```

Comment:
```suggestion
ssize_t win32_tape_device::d_read(int table_index, void* buffer, size_t count)
```

I think this is a more fitting name. Do you know why the `+3` / `-3` stuff is done ? Is it "just" for better error detection ( so that 0/1 arent valid ids ) ?

Reply:
that is probably to avoid the first three filehandles (i.e. stdin, stdout, stderr)

Reply:
and also I disagree. That should *not* be renamed as the `int` doesn't carry the meaning that this is supposed to be a file-descriptor by itself, so it should at least contain `fd`.

Reply:
Maybe we are talking beside each other, but fd is not a file descriptor.  Its an index into a table of HANDLEs, this is why i proposed `table_index` as an alternative name.

While its true that behind the curtain filedescriptors are basically just an index into an array,  I would suspect that calling our own indices file descriptors is more confusing than it is helpful.

Reply:
I understand your point.
This is the implementation for the Windows Tape backend. Windows specifically does *not* use file descriptors to access tapes, but Bareos wants to pass a file descriptor around.
So while this is "just" and index to a table internally, the API that we implement mandates that it is a file descriptor.
And also, I don't see why this specific implementation of `d_read()` should have a signature that differs from the other 8 implementations.